# Aula 01 (2021-06-01)
- Importação da biblioteca pandas e criação do DataFrame

In [6]:
import pandas as pd
df = pd.read_csv('datasets/kc_house_data.csv')

## Há duplicatas de ids?
- Os IDs são valores únicos para cada casa, não sendo possível que existam mais de uma casa com mesmo ID. 
- Para fazer essa verificação, a biblioteca pandas possui um método que facilita esse processo

In [12]:
df['id'].duplicated().value_counts()

False    21436
True       177
Name: id, dtype: int64

- Com isso, verificamos que existem 177 duplicatas e 21436 IDs únicos;
- É possível que as duplicatas existam porque o preço de alguns imóveis foram atualizados e os valores antigos não foram deletados.

In [15]:
df.drop_duplicates(subset='id', keep='last', inplace=True)

## Questão 1: Quantas casas estão disponíveis para compra?

In [17]:
df.shape[0]

21436

**Resposta:** 21436 imóveis estão disponíveis para compra.

## Questão 2: Quantos atributos os imóveis possuem?

In [18]:
df.shape[1]

21

**Resposta:** 21 atributos.

## Questão 3: Quais são esses atributos?

In [21]:
print(', '.join(df.columns))

id, date, price, bedrooms, bathrooms, sqft_living, sqft_lot, floors, waterfront, view, condition, grade, sqft_above, sqft_basement, yr_built, yr_renovated, zipcode, lat, long, sqft_living15, sqft_lot15


**Resposta:** Os atributos são: id, date, price, bedrooms, bathrooms, sqft_living, sqft_lot, floors, waterfront, view, condition, grade, sqft_above, sqft_basement, yr_built, yr_renovated, zipcode, lat, long, sqft_living15, sqft_lot15

## Questão 4: Qual é o ID da casa mais cara?

In [22]:
df.loc[df['price'] == max(df['price']),'id'].item()

6762700020

**Resposta:** O ID da casa mais cara é 6762700020

## Questão 5: Qual o ID da casa com maior número de quartos?

In [23]:
df.loc[df['bedrooms'] == max(df['bedrooms']),'id'].item()

2402100895

**Resposta:** O ID da casa com maior número de quartos é 2402100895

## Questão 6: Qual a soma de todos os quartos dos imóveis a venda?

In [24]:
df.loc[:,'bedrooms'].sum()

72273

**Resposta:** A soma de quartos é de 72273

## Questão 7: Número de casas com 2 banheiros

In [25]:
df.loc[df['bathrooms'] == 2,'id'].value_counts().sum()

1913

**Resposta:** 1913 imóveis possuem 2 banheiros

## Questão 8: Qual o preço médio dos imóveis?

In [26]:
df['price'].mean()

541649.962726255

**Resposta:** O preço médio é de US$541649.96

## Questão 9: Qual o preço médio de casas com 2 banheiros?

In [27]:
df.loc[df['bathrooms'] == 2, 'price'].mean()

459307.013591218

**Resposta:** O preço médio de casas com 2 banheiros é de US$459307.01

## Questão 10: Qual o preço mínimo de imóveis com 3 quartos?

In [28]:
df.loc[df['bedrooms'] == 3, 'price'].min()

89000.0

**Resposta:** O preço mínimo para imóveis com 3 quartos é de US$89000.00

## Questão 11: Quantas casas possuem mais de 300 metros quadrados (m²)?

- Pra obter o resultado, inicialmente é necessário realizar a conversão de pés quadrados (sqft) para metros quadrados. Para isso foi criada a função sqfeet2sqmeter.

In [33]:
def sqfeet2sqmeter(area):
    return area/(3.28084**2)

In [34]:
df['sqm_living'] = sqfeet2sqmeter(df['sqft_living'][:])
df.loc[df['sqm_living'] > 300, 'sqm_living'].value_counts().sum()

2251

**Resposta:** 2251 imóveis possuem mais de 300 m²

## Questão 12: Quantas casas possuem mais de 2 andares?

In [35]:
df.loc[df['floors'] > 2, 'floors'].value_counts().sum()

780

**Resposta:** 780 imóveis possuem mais de 2 andares.

## Questão 13: Quantas casas tem vista para o mar?

In [37]:
df.loc[df['waterfront'] != 0, 'waterfront'].value_counts().item()

163

**Resposta:** 163 imóveis possuem vista para o mar.

## Questão 14: Quantas casas possuem vista para o mar e 3 quartos?

In [39]:
df.loc[(df['waterfront'] != 0) & (df['bedrooms'] == 3), 'id'].value_counts().sum()

64

**Resposta:** 64 imóveis possuem vista para o mar e 3 quartos.

## Questão 15: Quantas casas possuem mais de 300 m² de sala de estar e mais de 2 banheiros?

In [40]:
df.loc[(df['sqm_living'] > 300) & (df['bathrooms'] > 2), 'id'].value_counts().sum()

2194

**Resposta:** 2194 casas possuem mais de 300 m² de sala de estar e mais de 2 banheiros.